packages/abi/src/matchers/sway-type-matchers.ts

```diff
@@ -105,7 +111,7 @@ export function createMatcher<T>(mappings: Record<SwayType, T>) {
     for (const [key, matcher] of swayTypeMatcherEntries) {
       if (matcher(swayType)) {
         const mapping = mappings[key as SwayType];
-        if (mapping) {
+        if (mapping !== undefined) {
           return mapping;
         }
```

Comment:
```suggestion
        if (key in mappings) {
          return mappings[key as SwayType];
        }
```

Would this be better?